swupd/rename.go

```diff
@@ -106,7 +106,15 @@ func trimRenamed(a []*File) []*File {
 		// and it is helpful to be able to ship just the differences for them.
 		// Worst case is that the files do not exist on the target system, in which
 		// case the swupd-client will fall back to doing a full download.
-		if f.DeltaPeer == nil {
+		//
+		// Renames are only supported for TypeFile right now so skip if the file is
+		// not the right type.
+		//
+		// Unfortunately deleted files have lost their association with a file type
+		// so we have to leave all deleted files in the list
+		// TODO: refactor to leave file type in the deleted file records. There is no
+		// reason to clear them.
+		if f.DeltaPeer == nil && (!f.Present() || f.Type == TypeFile) {
```

Comment:
Looks correct to me.

Optional: because we use trimRenamed more than once in the function, we are performing redundant checks. In the second time we use, we are calling it composed with trimSmall, reallocating stuff twice. I'd consider renaming (!) trimSmall and include the DeltaPeer check there, but not this check, that would be on the first trim call.